api/customers/CustomersApi.php

```diff
@@ -0,0 +1,150 @@
+        $this->primary_registrant = $primary_registrant;
+    }
+
+
+    /**
+     * Searches the customer profiles associated with a Square account using a supported query filter.
+     *
+     * @param       $filter
+     * @param array $sort
+     * @param int   $limit
+     * @return array
+     */
+    public function search($filter, array $sort = ['field' => 'CREATED_AT'], int $limit = 1): array
+    {
+        $search_parameters = [
+            'query' => [
+                'filter' => $filter,
+                'sort'   => $sort,
+            ],
+            'limit' => $limit,
+        ];
+        // Submit the search request.
+        $response = $this->api->sendRequest($search_parameters, $this->post_url . 'search');
+        // Do we have an error ?
+        if (is_array($response) && isset($response['error'])) {
+            return $response;
+        }
+        if (! $response || ! isset($response->customers)) {
+            // Square return an empty body if nothing is found, so we return an empty array.
+            return [];
+        }
+        // Got customer/s, return it.
+        return $response->customers;
+    }
+
+
+    /**
+     * Searches for the customer profile by his email.
+     *
+     * @param string $email
+     * @return array
+     */
+    public function findByEmail(string $email): array
+    {
+        $filter['email_address'] = ['exact' => $email];
+        return $this->search($filter);
+    }
+
+
+    /**
+     * Creates a new customer for a business.
+     *
+     * @return Object|array
+     */
+    public function create()
+    {
+        try {
+            $country_iso = EED_SquareOnsite::getCountryIsoByName($this->billing_info['country']);
+        } catch (\EE_Error | \ReflectionException $e) {
+            $country_iso = '';
+        }
+        $parameters = [
+            'given_name'    => $this->billing_info['first_name'] ?? '',
+            'family_name'   => $this->billing_info['last_name'] ?? '',
+            'email_address' => $this->billing_info['email'] ?? '',
+            'address' => [
+                'address_line_1' => $this->billing_info['address'] ?? '',
+                'address_line_2' => $this->billing_info['address2'] ?? '',
+                'locality'       => $this->billing_info['city'] ?? '',
+                'administrative_district_level_1' => $this->billing_info['state'] ?? '',
+                'postal_code'    => $this->billing_info['zip'] ?? '',
+                'country'        => $country_iso ?? '',
+            ],
+            'phone_number' => $this->billing_info['phone'] ?? '',
+            'reference_id' => (string) $this->primary_registrant->attendee_ID(),
+            'note'         => 'An event attendee'
+        ];
+
+        // Submit the create request.
+        $response = $this->api->sendRequest($parameters, $this->post_url);
+        // Do we have an error ?
+        if (is_array($response) && isset($response['error'])) {
```

Comment:
this conditional is repeated in the `search()` method and could benefit from being extracted.

for example:

```private function isErrorResponse($response) 
{
	return is_array($response) && isset($response['error']);
}
```

usage:

```suggestion
        if ($this->isErrorResponse($response)) {
```

the benefits being:

- slightly better readability
- if any change to error detection is required, it can be done in that one function but cover ALL use cases

Reply:
The issue is, that `isErrorResponse()` adds errors to the list which will throw an error on the payment itself, but in some cases we need the Payment to process even if there were errors in particular requests. Like - a payment shouldn't be marked as failed and can be processed without a new customer, imo.

Reply:
> The issue is, that isErrorResponse() adds errors to the list 

???

that was a new function I was suggesting that would simply replace the conditional that is duplicated in `search()` and `create()`

is there an existing function named `isErrorResponse()`??? If so, i didn't mean to use that one

Reply:
Sorry, don't mind that comment. Wrong place. Didn't remove it in time...